### PR TITLE
android fix: use correct instance for Unit classes getter

### PR DESCRIPTION
**Pull request by eszkadev**
On android we had issue with not working release builds
(worked when optimization flag -O0 was used for Unit.cpp).

Global which stored instance of UnitBase subtype was returned
after cast to a choosen subtype. Unfortunately in case when
android didn't work UnitWSD was returned in UnitKit::get() !

Thich patch makes us sure we don't cast to not-correct type.

### CODE REVIEW

File: common/Unit.cpp [hunk 1]
```diff
@@ -185,6 +206,12 @@ void UnitWSD::lookupTile(int part, int width, int height, int tilePosX, int tile
     }
 }
 
+UnitWSD& UnitWSD::get()
+{
+    assert(GlobalWSD && GlobalWSD->_type == UnitType::Wsd);
```
Comment (Ashod) [inline common/Unit.cpp:211]: If we guarantee that GlobalWSD always holds an instances of UnitWSD, why do we need to store the type and compare it?

File: common/Unit.cpp [hunk 2]
```diff
@@ -237,6 +280,11 @@ void UnitBase::returnValue(int &retValue)
     if (TimeoutThread.joinable())
         TimeoutThread.join();
 
+    if (Global == GlobalKit)
+        GlobalKit = nullptr;
+    if (Global == GlobalWSD)
+        GlobalWSD = nullptr;
```
Comment (Ashod) [inline common/Unit.cpp:286]: If we have two instances in the same process, surely we want to delete both. The `delete Global;` below will only delete one.

File: common/Unit.cpp [hunk 3]
```diff
@@ -24,6 +24,8 @@
 #include <common/SigUtil.hpp>
 
 UnitBase *UnitBase::Global = nullptr;
+UnitKit *GlobalKit = nullptr;
+UnitWSD *GlobalWSD = nullptr;
```
Comment (Ashod) [inline common/Unit.cpp:28]: If we have two separate variables for each type, why do we need the `UnitBase::Global`? Isn't it better to replace `UnitBase::Global` with `UnitBase::GlobalKit` and `UnitBase::GlobalWSD` ?

**Review by Ashod — APPROVED** [review]
Thanks Szymon! I left some comments for a better approach, but if this works, then let's get it in.

File: common/Unit.cpp (hunk 3 above)
Comment (eszkadev) [reply common/Unit.cpp:28]: we need UnitBase::Global for UnitBase::get() which we use for example in WebSocketHandler.cpp where we don't know which kind of test we run (it returns the last set UnitBase subclass)

**Pull request merged**